swift/apple/FirezoneKit/Sources/FirezoneKit/Helpers/LogCompressor.swift

```diff
@@ -9,91 +9,116 @@ import AppleArchive
 import Foundation
 import System
 
-struct LogCompressor {
-  private let destinationURL: URL?
-  let fileName: String
-
-  init(destinationURL: URL? = nil) {
-    let dateFormatter = ISO8601DateFormatter()
-    dateFormatter.formatOptions = [.withFullDate, .withTime, .withTimeZone]
-    let timeStampString = dateFormatter.string(from: Date())
-    self.fileName = "firezone_logs_\(timeStampString).aar"
-    self.destinationURL = destinationURL
+/// This module handles the business work of interacting with the AppleArchive framework to do the actual
+/// compression. It's used from both the app and tunnel process in nearly the same way, save for how the
+/// writeStream is opened.
+///
+/// In the tunnel process, the writeStream is a custom stream derived from our TunnelArchiveByteStream,
+/// which keeps state around the writing of compressed bytes in order to handle sending chunks back to the
+/// app process.
+///
+/// In the app process, the writeStream is derived from a passed file path where the Apple Archive
+/// framework handles writing for us -- no custom byte stream instance is needed.
+///
+/// Once the writeStream is opened, the remaining operations are the same for both.
+public struct LogCompressor {
+  enum CompressionError: Error {
+    case unableToReadSourceDirectory
+    case unableToOpenWriteStream
+    case unableToOpenCompressionStream
+    case unableToOpenEncodeStream
+    case unableToDefineHeaderKeys
   }
 
-  public func compressFolder(destinationURL: URL? = nil) async throws {
-    try await compressFolderReturningURL(destinationURL: destinationURL)
-  }
+  public init() {}
 
-  @discardableResult
-  public func compressFolderReturningURL(destinationURL: URL? = nil) async throws -> URL? {
-    guard let logFilesFolderURL = SharedAccess.logFolderURL,
-      let logFilesFolderPath = FilePath(logFilesFolderURL)
+  public func start(
+    source directory: URL,
+    to file: URL
+  ) throws {
+    guard let path = FilePath(file),
```

Comment:
Multiple variables per guard might be convenient to reuse the `else` clause but it messes with the formatting here. Can we extract a helper for `ArchiveByteStream.fileStream` so that the formatter puts these neatly on one line perhaps?

Alternatively, we could defer the creation of a `FilePath to the caller of this function.